src/shopify/theme_client.go

```diff
@@ -361,20 +359,28 @@ func (c Client) assetPath(query map[string]string) string {
 	return formatted
 }
 
-func unmarshalResponse(body io.ReadCloser, data interface{}) error {
-	reqBody, err := ioutil.ReadAll(body)
+func unmarshalResponse(resp *http.Response, data interface{}) error {
+	reqBody, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return ErrMalformedResponse
+		return fmt.Errorf("could not read response body on a response with HTTP status %v. This may mean that the request was not able to finish successfully", resp.StatusCode)
 	}
-	err = body.Close()
+	err = resp.Body.Close()
 	if err != nil {
 		return err
 	}
 	var re reqErr
-	mainErr := json.Unmarshal(reqBody, data)
-	basicErr := json.Unmarshal(reqBody, &re)
+	mainErr := json.Unmarshal(reqBody, data) // check if we can unmarshal into the expected returned data
+	basicErr := json.Unmarshal(reqBody, &re) // if not returned data, check if we can get errors from the body
 	if mainErr != nil && basicErr != nil {
-		return ErrMalformedResponse
+		errStr := "could not unmarshal JSON from response body on a response with HTTP status %v. This usually means themekit received an html error page."
+		tmpFile, err := ioutil.TempFile(os.TempDir(), "themekit-response-*.txt")
+		if err == nil {
+			tmpFile.Write([]byte(reqBody))
+			tmpFile.Close()
+			errStr += " Please find the full response at %v and include it with your ticket on github."
```

Comment:
Might this contain semi-sensitive data such as store names? We could suggest the people can edit it before posting?

Reply:
No, generally these error pages will have only errors in them. They *might* have shop domain in them but that is not considered sensitive.